DisguiseUnityRenderStream/Runtime/DisguiseRenderStream.cs

```diff
@@ -432,6 +394,89 @@ protected void AwaitFrame()
+        void OnBeginContextRendering(ScriptableRenderContext context, List<Camera> cameras)
+        {
+            if (!HasNewFrameData)
+                return;
+            
+            if (LatestFrameData.scene >= m_Schema.scenes.Length)
+                return;
+        
+            var spec = m_Schema.scenes[LatestFrameData.scene];
+            var images = m_SceneFields[LatestFrameData.scene].images;
+            if (images == null)
+                return;
+
+            // Only run once per frame for the main render context
+            if (m_HasUpdatedLiveTexturesThisFrame)
+                return;
+            foreach (var camera in cameras)
+            {
+                if (camera.cameraType != CameraType.Game)
+                    return;
+            }
+            m_HasUpdatedLiveTexturesThisFrame = true;
+            
+            var nImageParameters = spec.parameters.Count(t => t.type == RemoteParameterType.RS_PARAMETER_IMAGE);
+
+            using var imageData = new NativeArray<ImageFrameData>(nImageParameters, Allocator.Temp);
+            if (PluginEntry.instance.GetFrameImageData(spec.hash, imageData.AsSpan()) != RS_ERROR.RS_ERROR_SUCCESS)
+                return;
+
+            var i = 0;
+            foreach (var field in images)
+            {
+                if (field.GetValue() is RenderTexture renderTexture)
+                {
+                    // We may be temped to use RenderTexture instead of Texture2D for the scratch textures.
+                    // RenderTextures are always stored as typeless texture resources though, which aren't supported
+                    // by CUDA interop (used by Disguise under the hood):
+                    // https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__D3D11.html#group__CUDART__D3D11_1g85d07753780643584b8febab0370623b
+                    // Texture2D apply their GraphicsFormat to their texture resources.
+
+                    var texture = Texture2DPool.Instance.Get(new Texture2DDescriptor()
+                    {
+                        Width = (int)imageData[i].width,
+                        Height = (int)imageData[i].height,
+                        Format = PluginEntry.ToTextureFormat(imageData[i].format),
+                        Linear = true
+                    });
+
+                    NativeRenderingPlugin.InputImageData data = new NativeRenderingPlugin.InputImageData()
+                    {
+                        m_rs_getFrameImage = PluginEntry.instance.rs_getFrameImage_ptr,
+                        m_ImageId = imageData[i].imageId,
+                        m_Texture = texture.GetNativeTexturePtr()
+                    };
+
+                    if (NativeRenderingPlugin.InputImageDataPool.TryPreserve(data, out var dataPtr))
+                    {
+                        var cmd = CommandBufferPool.Get($"Receiving Disguise Image Parameter '{field.info.Name}'");
```

Comment:
Minor: string formatting allocates.

Reply:
Should I create a CommandBuffer outside the loop? It would contain the commands for all the images and avoid the string formatting.

Reply:
I like how you queue up all the commands into one buffer now. Not sure if it's more efficient now, but it certainly seems to be! 😄